libnmstate/nm/profile.py

```diff
@@ -21,9 +21,828 @@
+                                "mode, will do full deactivation before "
+                                "applying changes"
+                            )
+                            profile.to_deactivate_beforehand = True
+                            profiles_to_deactivate_beforehand.add(profile)
+                    profile.to_modify = True
+                    if (
+                        profile.iface_state[Interface.TYPE]
+                        in MASTER_IFACE_TYPES
+                    ):
+                        master_ifaces_to_edit.add(profile)
+                    else:
+                        ifaces_to_edit.add(profile)
+                elif profile.iface_state[Interface.STATE] in (
+                    InterfaceState.DOWN,
+                    InterfaceState.ABSENT,
+                ):
+                    affected_profiles = self._get_affected_profiles(
+                        profile, self._profiles
+                    )
+                    is_absent = (
+                        profile.iface_state[Interface.STATE]
+                        == InterfaceState.ABSENT
+                    )
+                    profile.to_deactivate = True
+                    profiles_to_deactivate.update(affected_profiles)
+                    if is_absent:
+                        profile.to_delete = True
+                        profiles_to_delete.update(affected_profiles)
+                    if (
+                        is_absent
+                        and profile.nmdev.is_software()
+                        and profile.nmdev.get_device_type()
+                        != NM.DeviceType.VETH
+                    ):
+                        profile.delete_dev = True
+                        devs_to_delete.add(profile)
+                else:
+                    raise NmstateValueError(
+                        "Invalid state {} for interface {}".format(
+                            profile.iface_state[Interface.STATE],
+                            profile.iface_state[Interface.NAME],
+                        )
+                    )
+
+        for profile in itertools.chain(master_ifaces_to_edit, ifaces_to_edit):
+            profile.to_deactivate = False
+            profile.to_delete = False
+            profile.delete_dev = False
+
+        return [
+            profiles_to_deactivate_beforehand,
+            profiles_to_delete,
+            new_master_not_enslaved,
+            new_ifaces_to_activate,
+            master_ifaces_to_edit,
+            new_ovs_port_to_activate,
+            new_ovs_interface_to_activate,
+            ifaces_to_edit,
+            new_vlan_x_to_activate,
+            profiles_to_deactivate,
+            profiles_to_delete,
+            devs_to_delete,
+        ]
+
+    def _prepare_state_for_profiles(self, net_state):
+        nm_dns.preapply_dns_fix_for_profiles(self._ctx, net_state)
+
+
+class NmProfile:
+    def __init__(self, context, save_to_disk, iface=None, iface_state=None):
```

Comment:
The `save_to_disk` should be argument of `create_proxy_profile()` or `apply_config()`.